dask_cuda/device_host_file.py

```diff
@@ -83,8 +111,11 @@ def __setitem__(self, key, value):
     def __getitem__(self, key):
         if key in self.host_buffer:
             obj = self.host_buffer[key]
-            del self.host_buffer[key]
-            self.device_buffer[key] = _deserialize_if_device(obj)
+            if isinstance(obj, Serialized):
+                del self.host_buffer[key]
+                self.device_buffer[key] = host_to_device(obj)
+            else:
+                return obj
```

Comment:
I'm curious if there is a way to offload these checks onto the `device_buffer` object entirely.  The buffer object is supposed to be able to get keys regardless of if they are in the fast or the slow storage.  So in theory I would hope that the `__getitem__` call here would look something like the following:

```python
def __getitem__(self, key):
    if key in self.device_buffer:
        return self.device_buffer[key]  # automatically calls host_to_device if necessary
    else:
        return self.host_buffer[key]  # automatically calls disk_to_host if necessary
```

I'm likely missing things here though.   For the plan Worker's storage, we do this by specifying functions that map both ways in the `Func` mapping:

https://github.com/dask/distributed/blob/65001f2d1c796f652bf77bbf222cf900bf1062b8/distributed/worker.py#L519-L523

Reply:
Indeed, you're right. Only the condition has to be inverted (`host_buffer` must come first). I added this in the latest commits too. Thanks for noticing.